charon/src/cli_options.rs

```diff
@@ -11,51 +11,50 @@ use std::path::PathBuf;
 // Note that because we need to transmit the options to the charon driver,
 // we store them in a file before calling this driver (hence the `Serialize`,
 // `Deserialize` options).
-#[derive(StructOpt, Default, Serialize, Deserialize)]
-#[structopt(name = "Charon")]
+#[derive(Parser, Default, Serialize, Deserialize)]
+#[clap(name = "Charon")]
 pub struct CliOpts {
     /// Extract the unstructured LLBC (i.e., don't reconstruct the control-flow)
-    #[structopt(long = "ullbc")]
+    #[clap(long = "ullbc")]
     #[serde(default)]
     pub ullbc: bool,
     /// Compile the package's library
-    #[structopt(long = "lib")]
+    #[clap(long = "lib")]
     #[serde(default)]
     pub lib: bool,
     /// Compile the specified binary
-    #[structopt(long = "bin")]
+    #[clap(long = "bin")]
     #[serde(default)]
     pub bin: Option<String>,
     /// Extract the promoted MIR instead of the built MIR
-    #[structopt(long = "mir_promoted")]
+    #[clap(long = "mir_promoted")]
     #[serde(default)]
     pub mir_promoted: bool,
     /// Extract the optimized MIR instead of the built MIR
-    #[structopt(long = "mir_optimized")]
+    #[clap(long = "mir_optimized")]
     #[serde(default)]
     pub mir_optimized: bool,
     /// Provide a custom name for the compiled crate (ignore the name computed
     /// by Cargo)
-    #[structopt(long = "crate")]
+    #[clap(long = "crate")]
     #[serde(default)]
     pub crate_name: Option<String>,
     /// The input file (the entry point of the crate to extract).
     /// This is needed if you want to define a custom entry point (to only
     /// extract part of a crate for instance).
-    #[structopt(long = "input", parse(from_os_str))]
+    #[clap(long = "input", parse(from_os_str))]
     #[serde(default)]
     pub input_file: Option<PathBuf>,
-    /// The destination directory, if we don't want to generate the output
-    /// .llbc files in the same directory as the input .rs files.
-    #[structopt(long = "dest", parse(from_os_str))]
+    /// The destination file. By default `<crate_name>.llbc`.
+    #[clap(long = "dest-file", parse(from_os_str))]
```

Comment:
I would like to keep a `--dest` or `--dest-dir`option: I think it convenient to be able to only specify the output directory (I often do it). You can have two options and check that at most one is set.